File: Linkedlist/csharp/GetNodeValue.cs
namespace LinkedListSolutions;

public static class GetNodeValue
{
    public static int? Solve(ListNode? head, int index)
    {
        var current = head;
        var position = 0;
        while (current is not null)
        {
            if (position == index)
            {
                return current.Value;
            }

            position++;
            current = current.Next;
        }

        return null;
    }
}

File: Linkedlist/java/getNodeValue.java
class GetNodeValue {
  public static Integer getNodeValue(ListNode head, int index) {
    ListNode current = head;
    int position = 0;
    while (current != null) {
      if (position == index) {
        return current.val;
      }
      position += 1;
      current = current.next;
    }
    return null;
  }

  public static void run() {
    // intentionally empty
  }
}
